docs/api.rst

```diff
@@ -1,93 +1,87 @@
+* :py:class:`bleak.BleakClient` an open connection to a BLE device
+* :py:class:`bleak.BleakGATTServiceCollection` the collection of services implemented in a BLE device
+* :py:class:`bleak.BleakGATTService` the name (or address) of a service implemented in a BLE device
+* :py:class:`bleak.BleakGATTCharacteristic` the name (or address) of a charactieristic (think: attribute or variable) implemented by such a service
+* :py:class:`bleak.BleakGATTDescriptor` extra properties of a charactieristic (think: user-readable name, or binary format)
 
-.. automodule:: bleak.backends.client
-    :members:
+You can then read and write characteristic values by calling methods of :py:class:`bleak.BleakClient`,
+passing the :py:class:`bleak.backends.service.BleakGATTCharacteristic` of the value you are interested in.
 
-Windows
-~~~~~~~
+Scanning
+--------
 
-.. automodule:: bleak.backends.winrt.client
-    :members:
+Use the top-level :py:class:`bleak.BleakScanner` object to scan for BLE 
+devices. The constructor may have 
+additional platform-specific arguments which you can find in the relevant 
+:doc:`apiimplementations` section. You can limit the scan to only look for a device with a specific name,
+or only for devices that implement a specific service.
 
-macOS
-~~~~~
+The scanner will return :py:class:`bleak.BLEDevice` objects.
 
-.. automodule:: bleak.backends.corebluetooth.client
-    :members:
+Scanner interface
+~~~~~~~~~~~~~~~~~
 
-Linux Distributions with BlueZ
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-.. automodule:: bleak.backends.bluezdbus.client
+.. autoclass:: bleak.BleakScanner
     :members:
+    :inherited-members:
 
-Python-for-Android/Kivy
-~~~~~~~~~~~~~~~~~~~~~~~
-
-.. automodule:: bleak.backends.p4android.client
-    :members:
-
-Scanning Clients
-----------------
-
-Interface
-~~~~~~~~~
-
-.. automodule:: bleak.backends.scanner
+.. autoclass:: bleak.BLEDevice
     :members:
+    :inherited-members:
 
-Windows
-~~~~~~~
-
-.. automodule:: bleak.backends.winrt.scanner
-    :members:
-
-macOS
-~~~~~
+Connecting
+----------
 
-.. automodule:: bleak.backends.corebluetooth.scanner
-    :members:
+After you find the correct :py:class:`bleak.BLEDevice` you will use the top-level :py:class:`bleak.BleakClient` object to connect to a BLE 
+device annd communicate with it. Alternatively, you can pass the BLE address of the device you want to connect to, and an automatic scan
+operation is performed.
 
-Linux Distributions with BlueZ
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Some of the methods, especially the constructor, may have 
+additional platform-specific arguments which you can find in the relevant 
+:doc:`apiimplementations` section.
 
-.. automodule:: bleak.backends.bluezdbus.scanner
-    :members:
 
-Python-for-Android/Kivy
-~~~~~~~~~~~~~~~~~~~~~~~
+Client interface
+~~~~~~~~~~~~~~~~
 
-.. automodule:: bleak.backends.p4android.scanner
+.. autoclass:: bleak.BleakClient
     :members:
+    :inherited-members:
 
+Addressing items in the BLE device
```

Comment:
`BleakGATTServiceCollection` and friends are more about enumerating rather than addressing or accessing. For devices with only one instance of each service/characteristics, using theses classes is not necessary unless you want to print meta info about the services/characteristics.

Reply:
ok.